src/test/java/com/johncnstn/auth/service/AuthServiceTest.java

```diff
@@ -0,0 +1,113 @@
+package com.johncnstn.auth.service;
+
+import com.johncnstn.auth.entity.UserEntity;
+import com.johncnstn.auth.entity.enums.UserRoleEntity;
+import com.johncnstn.auth.generated.model.SignInRequest;
+import com.johncnstn.auth.generated.model.User;
+import com.johncnstn.auth.generated.model.UserRole;
+import com.johncnstn.auth.repository.UserRepository;
+import com.johncnstn.auth.security.DomainGrantedAuthority;
+import com.johncnstn.auth.security.DomainUserDetails;
+import com.johncnstn.auth.security.JwtTokens;
+import com.johncnstn.auth.security.TokensProvider;
+import com.johncnstn.auth.service.impl.AuthServiceImpl;
+import com.johncnstn.auth.unit.AbstractUnitTest;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.util.UUID;
+
+import static java.lang.System.currentTimeMillis;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class AuthServiceTest extends AbstractUnitTest {
+
+    @Mock
+    private AuthenticationManager authenticationManager;
+
+    @Mock
+    private PasswordEncoder passwordEncoder;
+
+    @Mock
+    private TokensProvider tokensProvider;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Test
+    public void testSignUp() {
+        // GIVEN
+        var rawUser = new User();
+        rawUser.setEmail("test@mail.com");
+        rawUser.setPassword("demo123");
+
+        var entityToReturn = new UserEntity(
+                UUID.randomUUID(),
+                rawUser.getEmail(),
+                "xyz",
+                UserRoleEntity.USER);
```

Comment:
UserRoleEntity import

Reply:
There are three constants with the name USER in this class.